include/opt-sched/Scheduler/enumerator.h

```diff
@@ -1513,7 +1525,32 @@ inline void TreeNodeAllocWrapper::Reset() {
 
 /****************************************************************************/
 
+inline ReadyList *ReadyListAllocWrapper::Alloc(DataDepGraph* dataDepGraph,
+                                               SchedPriorities prirts, int solverID) {
+  ReadyList *rdyLst;
+  rdyLst = allctr_->GetObject();
+  rdyLst->Reset();
```

Comment:
By reworking the underlying arena allocator of the priority list, we should be able to "reset" by simply reseting the value of a pointer, rather than iterating through the entire free list and deallocating each element.